opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java

```diff
@@ -92,23 +112,56 @@ public Iterator<ExprValue> iterator() {
         return (ExprValue) ExprTupleValue.fromExprValueMap(builder.build());
       }).iterator();
     } else {
+      boolean includeId = includes.stream().anyMatch(
+          include -> include.equalsIgnoreCase(METADATA_FIELD_ID)
+      );
+      boolean includeIndex = includes.stream().anyMatch(
+          include -> include.equalsIgnoreCase(METADATA_FIELD_INDEX)
+      );
+      boolean includeScore = includes.stream().anyMatch(
+          include -> include.equalsIgnoreCase(METADATA_FIELD_SCORE)
+      );
+      boolean includeMaxScore = includes.stream().anyMatch(
+          include -> include.equalsIgnoreCase(METADATA_FIELD_MAXSCORE)
+      );
+      boolean includeSort = includes.stream().anyMatch(
+          include -> include.equalsIgnoreCase(METADATA_FIELD_SORT)
+      );
+      ExprFloatValue maxScore = Float.isNaN(hits.getMaxScore())
+          ? null : new ExprFloatValue(hits.getMaxScore());
       return Arrays.stream(hits.getHits())
           .map(hit -> {
-            ExprValue docData = exprValueFactory.construct(hit.getSourceAsString());
-            if (hit.getHighlightFields().isEmpty()) {
-              return docData;
-            } else {
-              ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
-              builder.putAll(docData.tupleValue());
+            String source = hit.getSourceAsString();
+            ExprValue docData = exprValueFactory.construct(source);
+
+            ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
+            builder.putAll(docData.tupleValue());
+            if (includeIndex) {
```

Comment:
`includeIndex` (and others too) is used only once. Maybe do `...anyMatch()` right here?
Or make an auxiliary function, so you can do
```suggestion
            if (include(METADATA_FIELD_INDEX)) {
```

Reply:
changed in ccdc290fa9fcbc40b6beeefe55f6d8e3ab93125b 
I moved everything to a list and removed these boolean checks.